src/io/GeoJSONWriter.cpp

```diff
@@ -39,111 +39,132 @@ using json = geos_nlohmann::ordered_json;
 namespace geos {
 namespace io { // geos.io
 
-std::string GeoJSONWriter::write(const geom::Geometry* geometry, GeoJSONType type) {
+std::string GeoJSONWriter::write(const geom::Geometry* geometry, GeoJSONType type)
+{
     json j;
     encode(geometry, type, j);
-    std::string geojson = j.dump();
+    const std::string& geojson = j.dump();
     return geojson;
 }
 
-std::string GeoJSONWriter::writeFormatted(const geom::Geometry* geometry, GeoJSONType type, int indent) {
+std::string GeoJSONWriter::writeFormatted(const geom::Geometry* geometry, GeoJSONType type, int indent)
+{
     json j;
     encode(geometry, type, j);
-    std::string geojson = j.dump(indent);
+    const std::string& geojson = j.dump(indent);
```

Comment:
Just for my curiousity (I'm no cxx expert) is this really saving a copy (assigning to a const reference before returning) or is it just making us feel good about apparently saving a copy?

Reply:
I have made a new  PR #467  to fix bug and enhance functions.